### PR TITLE
vim's C binding

### PR DESCRIPTION
**Pull request by zim0369**
I'm not sure why I see duplicate commits that you've already merged before. I'm not well versed in github.

I will try to work on `shift+s` for which I think I might need to write a function.

Thank you so much! 
Sorry for the duplicate commits. Not sure how github works. Feel free to squash them.

### CODE REVIEW

File: src/edit_mode/vi/command.rs [hunk 1]
```diff
@@ -137,6 +141,7 @@ pub enum Command {
     EnterViAppend,
     EnterViInsert,
     Undo,
+    ChangeTillEnd,
```
Comment (sholderbach) [inline src/edit_mode/vi/command.rs:144]: Would you mind making that consistent with `ToLineEnd` or `ToEnd`

File: src/edit_mode/vi/command.rs [hunk 2]
```diff
@@ -164,6 +169,7 @@ impl Command {
             Self::PasteAfter => vec![ReedlineOption::Edit(EditCommand::PasteCutBufferAfter)],
             Self::PasteBefore => vec![ReedlineOption::Edit(EditCommand::PasteCutBufferBefore)],
             Self::Undo => vec![ReedlineOption::Edit(EditCommand::Undo)],
+            Self::ChangeTillEnd => vec![ReedlineOption::Edit(EditCommand::CutToLineEnd)],
```
Comment (sholderbach) [inline src/edit_mode/vi/command.rs:172]: This should probably be `ClearToLineEnd`. `CutToLineEnd` will set the clipboard to the deleted content.

**Comment by sholderbach** [comment]
> Hey, seems like the function necessary for `shift+s` is already present. Probably one of these in [editor.rs](https://github.com/nushell/reedline/blob/2e2bdc54621643e7bee5ba2e2d9bf28e757074e0/src/core_editor/editor.rs#L56)
> 
> ```rust
> EditCommand::CutCurrentLine => self.cut_current_line(),
> EditCommand::CutFromStart => self.cut_from_start(),
> EditCommand::CutFromLineStart => self.cut_from_line_start(),
> ```
> 
> Which one?

`_current_line` affects the whole line.

**Comment by sholderbach** [comment]
> I'm not sure why I see duplicate commits that you've already merged before. I'm not well versed in github.
> 
> Sorry for the duplicate commits. Not sure how github works. Feel free to squash them.

We have a squash (or rebase) policy for PR merges, so you don't have to worry about your history too much.

Commiting to the main branch happens to all of us at some point :P Are you using git from a particular GUI or from the command line?

**Comment by herlon214** [comment]
Ok, everybody is too serious. Where's the 404 pull request not found joke? xD

**Comment by zim0369** [comment]
> > I'm not sure why I see duplicate commits that you've already merged before. I'm not well versed in github.
> > Sorry for the duplicate commits. Not sure how github works. Feel free to squash them.
> 
> We have a squash (or rebase) policy for PR merges, so you don't have to worry about your history too much.
> 
> Commiting to the main branch happens to all of us at some point :P Are you using git from a particular GUI or from the command line?

cli

**Pull request closed without merging**